pkg/issuer/acme/dns/webhook/webhook.go

```diff
@@ -149,7 +129,37 @@ func (r *Webhook) Initialize(kubeClientConfig *rest.Config, stopCh <-chan struct
 	return nil
 }
 
-func (r *Webhook) loadConfig(cfgJSON apiext.JSON) (*cmapi.ACMEIssuerDNS01ProviderWebhook, error) {
+func (r *Webhook) buildPayload(ch *v1alpha1.ChallengeRequest, action v1alpha1.ChallengeAction) (*rest.RESTClient, *v1alpha1.ChallengePayload, string, error) {
+	// create a copy just to be certain we don't modify something unexpectedly
+	req := ch.DeepCopy()
+
+	// extract the complete solver config, including groupName and solverName
+	cfg, err := loadConfig(*req.Config)
+	if err != nil {
+		return nil, nil, "", err
+	}
+
+	// obtain a REST client that can be used to communicate with the webhook
+	cl, err := r.restClientForGroup(cfg.GroupName)
+	if err != nil {
+		return nil, nil, "", err
+	}
+
+	// build the ChallengePayload resource
+	pl := &v1alpha1.ChallengePayload{
+		Request: req,
+	}
+	pl.Request.Action = action
+	// When using the webhook provider, the 'config' on the ChallengeRequest
+	// will be the complete marshaled configuration as specified on the issuer.
+	// Instead of passing all this extra config along, we instead extract out
+	// only the 'config' field and submit that to the webhook.
+	pl.Request.Config = cfg.Config
```

Comment:
@Diaphteiros this is ultimately the fix here. Instead of setting `.config` in the DNS package, we now overwrite it in the actual webhook DNS provider itself.